javaslang/src/main/java/javaslang/collection/Comparators.java

```diff
@@ -47,7 +47,11 @@ private Comparators() {
         long serialVersionUID = 1L;
 
         static <T> SerializableComparator<T> of(Comparator<T> comparator) {
-            return comparator::compare;
+            if (comparator instanceof SerializableComparator) {
```

Comment:
I needed this the existing `comparator` to create new instances of the `PriorityQueue`, but wanted it to be callable for normal `Comparator`s also, so the constructor accepted the Java one and converted in inside.